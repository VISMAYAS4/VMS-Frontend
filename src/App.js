import "bootstrap/dist/css/bootstrap.min.css";
import "./App.css";

import { Link, Route, Router, Switch } from "react-router-dom";
import React, { Component } from "react";

import AddBooking from "./components/bookingComponents/AddBooking";
import AddCustomer from './components/customerComponents/AddCustomer';
import AddPayment from './components/paymentComponents/AddPayment';
import AddUser from "./components/userComponents/AddUser"
import AddVehicle from "./components/vehicleComponents/AddVehicle";
import AppBar from '@material-ui/core/AppBar';
import Dashboard from "./components/Dashboard";
import DetailViewBooking from "./components/bookingComponents/DetailViewBooking";
import DetailViewCustomer from './components/customerComponents/DetailViewCustomer';
import DetailViewPayment from './components/paymentComponents/DetailViewPayment';
import DetailViewVehicle from "./components/vehicleComponents/DetailViewVehicle";
import Login from "./components/LoginComponent";
import Profile from "./components/ProfileComponent";
import Register from "./components/RegisterComponent";
import Toolbar from '@material-ui/core/Toolbar';
import TotalPaymentByVehicle from './components/paymentComponents/TotalPaymentByVehicle';
import TotalRevenue from './components/paymentComponents/TotalRevenue';
import Typography from '@material-ui/core/Typography';
import UpdateBooking from "./components/bookingComponents/UpdateBooking";
import UpdateCustomer from './components/customerComponents/UpdateCustomer';
import UpdateVehicle from "./components/vehicleComponents/UpdateVehicle";
import UserList from "./components/userComponents/UserList"
import ViewBookings from "./components/bookingComponents/ViewBookings";
import ViewCustomers from './components/customerComponents/ViewCustomers';
import ViewPayments from './components/paymentComponents/ViewPayments';
import ViewVehicles from "./components/vehicleComponents/ViewVehicles";
import { clearMessage } from "./actions/message";
import { connect } from "react-redux";
import { history } from './helpers/history';
import { logout } from "./actions/auth";

class App extends Component {
  constructor(props) {
    super(props);
    this.logOut = this.logOut.bind(this);

    this.state = {
      showModeratorBoard: false,
      showAdminBoard: false,
      currentUser: undefined,
    };

    history.listen((location) => {
      props.dispatch(clearMessage()); // clear message when changing location
    });
  }

  componentDidMount() {
    const user = this.props.user;

    if (user) {
      this.setState({
        currentUser: user,
        showAdminBoard: user.roles.includes("ROLE_ADMIN"),
      });
    }
  }

  logOut() {
    this.props.dispatch(logout());
  }

  render() {
    const { currentUser, showAdminBoard } = this.state;

    return (
      <Router history={history}>
        <div>
          <AppBar position="static">
            <Toolbar>
              {/* <HomeIcon style={{ marginRight: "10px" }}/> */}
              <Typography variant="h6"><Link to={"/"} style={{ color: 'white' }}>
                Vehicle Management Application
            </Link></Typography>

              {showAdminBoard && <Typography style={{ margin: "10px" }}>
                <Link to={"/users"} style={{ color: 'white' }}>Users</Link></Typography>}
              <Typography style={{ margin: "10px" }}>
                <Link to={"/viewCustomers"} style={{ color: 'white' }}>Customers</Link></Typography>
              <Typography style={{ margin: "10px" }}>
                <Link to={"/viewBooking"} style={{ color: 'white' }}>Bookings</Link></Typography>
              <Typography style={{ margin: "10px" }}>
                <Link to={"/viewVehicles"} style={{ color: 'white' }}>Vehicles</Link></Typography>
              <Typography style={{ margin: "10px" }}>
                <Link to={"/viewPayment"} style={{ color: 'white' }}>Payments</Link></Typography>

              {currentUser ? (
                <div style={{ marginLeft: 'auto', display: 'flex' }}>
                  <Typography variant="h6" style={{ margin: "10px" }}>
                    <Link to={"/profile"} style={{ color: 'white' }}>
                      {currentUser.username}
                    </Link>
                  </Typography>
                  <Typography variant="h6" style={{ margin: "10px" }}>
                    <a href="/login" style={{ color: 'white' }} onClick={this.logOut}>
                      LogOut
                  </a>
                  </Typography>
                </div>
              ) : (
                  <div style={{ marginLeft: 'auto', display: 'flex' }}>
                    <Typography variant="h6" style={{ margin: "10px" }}>
                      <Link to={"/login"} style={{ color: 'white' }}>
                        Login
                  </Link>
                    </Typography>

                    <Typography variant="h6" style={{ margin: "10px" }}>
                      <Link to={"/register"} style={{ color: 'white' }}>
                        SignUp
                  </Link>
                    </Typography>
                  </div>
                )}
            </Toolbar>
          </AppBar>

          <div>
            <Switch>
              <Route exact path={["/", "/home"]} component={Login} />
              <Route exact path="/login" component={Login} />
              <Route path="/users" component={UserList}></Route>
              <Route path="/add-user/:id" component={AddUser}></Route>
              <Route exact path="/register" component={Register} />
              <Route exact path="/profile" component={Profile} />
              <Route path="/dashboard" component={Dashboard} />
              <Route path="/viewBooking"><ViewBookings /></Route>
              <Route path="/addBooking"><AddBooking /></Route>
              <Route path="/updateBooking/:id" component={UpdateBooking} />
              <Route path="/detailViewBooking/:id" component={DetailViewBooking} />
              <Route path="/viewCustomers"><ViewCustomers /></Route>
              <Route path="/addCustomer"><AddCustomer /></Route>
              <Route path="/updateCustomer/:id" component={UpdateCustomer} />
              <Route path="/detailViewCustomer/:id" component={DetailViewCustomer} />
              <Route path="/viewPayment"><ViewPayments /></Route>
              <Route path="/addPayment"><AddPayment /></Route>
              <Route path="/viewTotalRevenue"><TotalRevenue /></Route>
              <Route path="/detailViewPayment/:id" component={DetailViewPayment} />
              <Route path="/viewTotalPaymentByVehicle"><TotalPaymentByVehicle /></Route>
              <Route path="/viewVehicles"><ViewVehicles /></Route>
              <Route path="/addVehicle"><AddVehicle /></Route>
              <Route path="/updateVehicle/:id" component={UpdateVehicle} />
              <Route path="/detailViewVehicle/:id" component={DetailViewVehicle} />
            </Switch>
          </div>
        </div>
      </Router>
    );
  }
}

function mapStateToProps(state) {
  const { user } = state.auth;
  return {
    user,
  };
}

export default connect(mapStateToProps)(App);
