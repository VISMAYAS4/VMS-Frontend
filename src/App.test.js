import App from './App';
import { Provider } from 'react-redux';
import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);
 
describe('Render App when not logged in', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
        auth : {user:{
          accessToken: "",
          email: "taniya@gmail.com",
          id: 1,
          roles: [
            "ROLE_ADMIN"
          ],
          tokenType: "Bearer",
          username: "Taniya"
        }, 
      isLoggedIn : false},
      message : ""
    });
 
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
 
});

describe('Render App when logged in', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
        auth : {user:{
          accessToken: "",
          email: "taniya@gmail.com",
          id: 1,
          roles: [
            "ROLE_ADMIN"
          ],
          tokenType: "Bearer",
          username: "Taniya"
        }, 
      isLoggedIn : true},
      message : ""
    });
 
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
 
});