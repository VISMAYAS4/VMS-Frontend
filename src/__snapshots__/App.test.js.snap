// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App when logged in should render with given state from Redux store 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Vehicle Management Application
        </a>
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/users"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Users
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewCustomers"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Customers
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewBooking"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Bookings
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/dashboard"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Vehicles
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewPayment"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Payments
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewActiveBooking"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Active Bookings
        </a>
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "marginLeft": "auto",
          }
        }
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          style={
            Object {
              "margin": "10px",
            }
          }
        >
          <a
            href="/profile"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            Taniya
          </a>
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-h6"
          style={
            Object {
              "margin": "10px",
            }
          }
        >
          <a
            href="/login"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            LogOut
          </a>
        </h6>
      </div>
    </div>
  </header>
  <div>
    <main
      className="MuiGrid-root Profile-root-11 MuiGrid-container"
    >
      <div
        className="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
        style={
          Object {
            "margin": "auto",
            "marginTop": "10%",
          }
        }
      >
        <div
          className="Profile-paper-13"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle Profile-avatar-14 MuiAvatar-colorDefault"
          >
            T
          </div>
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Profile
          </h5>
          <br />
          <br />
          <div
            className="Profile-row-15"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6"
            >
              Username:
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6"
            >
              Taniya
            </h6>
          </div>
          <div
            className="Profile-row-15"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6"
            >
              Email:
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6"
            >
              taniya@gmail.com
            </h6>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Render App when not logged in should render with given state from Redux store 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Vehicle Management Application
        </a>
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/users"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Users
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewCustomers"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Customers
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewBooking"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Bookings
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/dashboard"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Vehicles
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewPayment"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Payments
        </a>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <a
          href="/viewActiveBooking"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Active Bookings
        </a>
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "marginLeft": "auto",
          }
        }
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          style={
            Object {
              "margin": "10px",
            }
          }
        >
          <a
            href="/profile"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            Taniya
          </a>
        </h6>
        <h6
          className="MuiTypography-root MuiTypography-h6"
          style={
            Object {
              "margin": "10px",
            }
          }
        >
          <a
            href="/login"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            LogOut
          </a>
        </h6>
      </div>
    </div>
  </header>
  <div>
    <div
      className="container mt-3 "
    >
      <main
        className="MuiGrid-root Login-root-1 MuiGrid-container"
      >
        <div
          className="MuiGrid-root Login-image-2 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
        />
        <div
          className="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
        >
          <div
            className="Login-paper-3"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle Login-avatar-4 MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
            <h1
              className="MuiTypography-root MuiTypography-h5"
            >
              Sign in
            </h1>
            <form
              className="Login-form-5"
              onSubmit={[Function]}
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                validations={
                  Array [
                    [Function],
                  ]
                }
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="username"
                  id="username-label"
                >
                  Username
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="username"
                    autoFocus={true}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="username"
                    name="username"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-9"
                    >
                      <span>
                        Username
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                validations={
                  Array [
                    [Function],
                  ]
                }
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="password"
                  id="password-label"
                >
                  Password
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="current-password"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="password"
                    name="password"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-9"
                    >
                      <span>
                        Password
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained Login-submit-6 MuiButton-containedPrimary MuiButton-fullWidth"
                data-testid="login-button"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="submit"
              >
                <span
                  className="MuiButton-label"
                >
                  Sign In
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
              <div
                className="MuiGrid-root MuiGrid-container"
              >
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <a
                    href="/register"
                  >
                    Don't have an account? Sign Up
                  </a>
                </div>
              </div>
              <button
                disabled={false}
                style={
                  Object {
                    "display": "none",
                  }
                }
              />
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
